app-frontend/src/app/components/pages/project/layer/aoi/index.js

```diff
@@ -1,11 +1,286 @@
 import tpl from './index.html';
+import _ from 'lodash';
+
+const mapName = 'project';
+const mapLayerName = 'Project Layer';
+const geomDrawType = 'Polygon';
+const defaultColor = '#738FFC';
 
 class ProjectLayerAoiController {
+    constructor(
+        $rootScope, $scope, $state, $log, $window,
+        uuid4, moment, projectService, paginationService, mapService,
+        modalService, shapesService
+    ) {
+        'ngInject';
+        $rootScope.autoInject(this, arguments);
+    }
+
+    $onInit() {
+        this.selected = new Map([]);
+        this.mapId = mapName;
+        this.geomDrawType = geomDrawType;
+        this.filterList = [
+            {
+                name: 'Project AOIs',
+                id: this.uuid4.generate(),
+                type: 'AOI'
+            },
+            {
+                name: 'Vector Data',
+                id: this.uuid4.generate(),
+                type: 'VECTOR'
+            }
+        ];
+        this.selectedGeom = null;
+        this.setMapLayers();
+        this.projectService
+            .getProjectLayer(this.project.id, this.layer.id)
+            .then(layer => {
+                this.layer = layer;
+                this.layerGeom = this.layer.geometry;
+            })
+            .catch(err => {
+                this.$log.error(err);
+            });
+        this.dataType = this.filterList[0].type;
+        this.fetchPage(1, this.filterList[0].id);
+        this.displayAoiSource = this.filterList[0].name;
+    }
+
+    $onDestroy() {
+        this.removeMapLayers();
+    }
+
+    getMap() {
+        return this.mapService.getMap(mapName);
+    }
+
+    setMapLayers() {
+        let mapLayer = this.projectService.mapLayerFromLayer(this.project, this.layer);
+        return this.getMap().then(map => {
+            map.setLayer(mapLayerName, mapLayer, true);
+        });
+    }
+
+    removeMapLayers() {
+        this.getMap().then(map => {
+            map.deleteLayers(mapLayerName);
+        });
+    }
+
+    fetchPage(page = 1, typeId = this.filterList[0].id) {
+        this.itemList = [];
+        let fetchedList = [];
+        this.dataType = this.filterList.find(fl => fl.id === typeId).type;
+        const query = this.getQueryByType(page);
+
+        const currentQuery = query.then((paginatedResponse) => {
+            if (this.dataType === 'AOI') {
+                fetchedList = paginatedResponse.results;
+            } else {
+                fetchedList = paginatedResponse.features;
+            }
+
+            this.itemList = _.compact(fetchedList.map(fl => {
+                if (this.dataType === 'AOI' && !fl.geometry) {
+                    return 0;
+                }
+                return this.createItemInfo(fl);
+            }));
+
+            if (this.itemList.length && this.dataType === 'AOI') {
```

Comment:
If the user has only one vector, when the filter is changed to vector the first item is automatically selected and can't be unselected. If you then delete the AOI, the vector is still selected and can't be put back on the map.